src/legacy/core_plugins/kibana/public/visualize_embeddable/visualize_embeddable_factory.tsx

```diff
@@ -129,14 +127,27 @@ export class VisualizeEmbeddableFactory extends EmbeddableFactory<
     });
   }
 
+  public getSavedVisualizations() {
+    if (!this.savedVisualizations) {
+      const services = {
+        savedObjectsClient: npStart.core.savedObjects.client,
+        indexPatterns: npStart.plugins.data.indexPatterns,
+        chrome: npStart.core.chrome,
+        overlays: npStart.core.overlays,
+      };
+      this.savedVisualizations = createSavedVisLoader(services);
```

Comment:
is there a reason we are creating a new savedVisLoader ? i think we should use the one that is already in the registry.
the registry should be exposed (i guess for now from kibana plugin, but in the long run we'll have to put it somewhere else). 

in the setup method we should expose a way to add to the registry and in the start method we should expose a way to read the registry.

then in visualizations plugin we can 
- add to the registry (saved_visualizations belongs into visualizations plugin imo)
- read the registry in for example embeddable factory

Reply:
We're removed the registry already, but we can definitely
improve this part of the code to avoid creating another savedVisLoader instance. I'll improve this part of the code, thx for your input 👍

Reply:
dear @ppisljar , I've refactored this, now there's only one savedVisLoader, and I've moved the `saved_vizualizations` folder to the `visualizations` plugin. no registry is necessary.